src/scripts/store.js

```diff
@@ -0,0 +1,62 @@
+export default {
+	state: {
+		activeIndex: 0,
+		maxIndex : 0,
+		activeMediaItem : {},
+		activeDomNode : null,
+		isLoading : false
+	},
+	getters : {
+		itemName (state) {
+			if (state.activeMediaItem.length === 0) {
+				return 'Loading ...';
+			}
+			return state.activeMediaItem.name;
+		},
+		itemType (state) {
+			if (!state.activeMediaItem.mimetype) {
+				return null;
+			}
+			return state.activeMediaItem.mimetype.split('/')[0];
+		},
+		image (state) {
+			return (state.activeDomNode.get(0) instanceof HTMLImageElement) ? state.activeDomNode.get(0) : false;
+		},
+		video (state) {
+			return (state.activeDomNode.get(0) instanceof HTMLVideoElement) ? state.activeDomNode.get(0) : false;
+		}
+	},
+	mutations: {
+		setActiveIndex (state, index) {
+			state.activeIndex = parseInt(index);
```

Comment:
best practice is `parseInt(index, 10)` to avoid issues with leading zeroes and octal.
also see other occurrences of `parseInt`

probably doesn't apply here